source/MaterialXRuntime/Private/PvtPrim.h

```diff
@@ -72,7 +72,9 @@ class PvtPrim : public PvtObject
 public:
     static PvtDataHandle createNew(const RtTypeInfo* typeInfo, const RtToken& name, PvtPrim* parent);
 
-    void dispose();
+    void dispose(bool state);
```

Comment:
Make this a settable state so we can bring prims back to life for undo/redo or create and remove commands. Instead, the new destroy() below will perform irreversible deletion of the prim, releasing all child prims and attributes etc.

Reply:
Just to check logic here: Now we keep all the previously existing prims when executed from commands or we do not?
If it's the former then don't we keep on using up memory for deleted prims until the undo queue is flushed? I assume it's the latter.

Reply:
Yes, the command will store a reference to the prim which makes the ref pointer not delete itself. So the prim can be restored in undo. This means that deleted prims will remain on the undo queue, and use up memory, until the queue is flushed.

There's a couple of reasons why I found this to be the best solution. First, a user will expect the full prim state to be restored by undo delete, including any values that have been changed or metadata created on the prim. If the prim is fully deleted we need to store this full state somehow anyway, which would use up memory anyway. plus it will be messy logic restoring the full state. Second, the prims are quite lightweight objects anyway, it's mainly just a set attributes with values. So the queue can grow very long until it would be a noticeable memory footprint. If it turns out to be a real problem in the future a simple solution would be to put a max length on the undo queue, so we have garbage collection on the history further back.

(and note that state is only saved when removing prims by command. If removing by explicit API calls to the stage the operation is not undoable and the prim is fully deleted directly).

Reply:
Good point. Best not to invent a way to store state. I wasn't sure about size of prims but if it's light(er) weight it makes sense. Thanks.